jsonrpc/src/main/scala/scala/meta/jsonrpc/Logger.scala

```diff
@@ -0,0 +1,45 @@
+// scalafmt: { maxColumn = 120 }
+package scala.meta.jsonrpc
+
+trait Logger {
+  def trace(message: => String)(implicit location: SourceLocation): Unit
+  def trace(message: => String, cause: Throwable)(implicit location: SourceLocation): Unit
+  def debug(message: => String)(implicit location: SourceLocation): Unit
+  def debug(message: => String, cause: Throwable)(implicit location: SourceLocation): Unit
+  def info(message: => String)(implicit location: SourceLocation): Unit
+  def info(message: => String, cause: Throwable)(implicit location: SourceLocation): Unit
+  def warn(message: => String)(implicit location: SourceLocation): Unit
+  def warn(message: => String, cause: Throwable)(implicit location: SourceLocation): Unit
+  def error(message: => String)(implicit location: SourceLocation): Unit
+  def error(message: => String, cause: Throwable)(implicit location: SourceLocation): Unit
+}
+
+object Logger {
+  object StdoutLogger extends Logger {
```

Comment:
I tried to keep bells and whistles to an absolute minimum while still having at least one out-of-the-box working logger that doesn't suck.